modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java

```diff
@@ -130,24 +129,26 @@ private Predicate _getPredicate(
 				continue;
 			}
 
-			String value = null;
+			Object value = entryValues.get(objectField.getName());
 
-			if (StringUtil.equals(
-					objectField.getBusinessType(),
-					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+			String stringValue = String.valueOf(value);
+
+			if (value == null) {
+				stringValue = null;
+			}
```

Comment:
I think you could replace this code to:
`String stringValue = GetterUtil.getString(value, null);`
Please, test your scenarios :+1:

Reply:
This would have a slightly different behavior. The `String.valueOf` will call `toString` if the value is not null. Whereas `GetterUtil.getString` will return the default value (null in this case) if the value is not an instance of string.